src/ProxyClient/Request/RequestQueue.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient\Request;
+
+use FOS\HttpCache\Exception\InvalidUrlException;
+use Http\Discovery\UriFactoryDiscovery;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Stores each invalidation request and replicates it over all HTTP cache servers
+ *
+ * @author David de Boer <david@ddeboer.nl>
+ */
+class RequestQueue implements \Countable
```

Comment:
I’ll leave it at `RequestQueue` for now, as it corresponds to `queueRequest()`. If we come up with something better later, we can always change it as long as we’re < 2.0.